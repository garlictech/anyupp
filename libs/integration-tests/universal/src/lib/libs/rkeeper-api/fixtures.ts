import * as CrudApi from '@bgap/crud-gql/api';
import {
  Dish,
  RKeeperBusinessEntityInfo,
  RkeeperFixtures,
} from '@bgap/rkeeper-api';
import { RequiredId } from '@bgap/shared/types';
import {
  unitFixture,
  productFixture,
  chainFixture,
  groupFixture,
  rkeeperEndpoint,
  yellowRestaurantId,
  yellowRkeeperUsername,
  yellowRkeeperPassword,
} from '@bgap/shared/fixtures';

const testIdPrefix = `rkeeper-cf0d1110-a2ce-45cf-aa69-6782bbc44cad`;

export const rawData = {
  data: {
    dishes: [
      RkeeperFixtures.dish,
      RkeeperFixtures.dish2,
      RkeeperFixtures.duplicatedDish,
      RkeeperFixtures.badDish,
      RkeeperFixtures.inactiveDish,
      RkeeperFixtures.dishWithModifier,
      RkeeperFixtures.dishWithSameModifier,
    ],
    modifiers: [RkeeperFixtures.modifier, RkeeperFixtures.unusedModifier],
  },
};

export const rkeeperProductGuid = 'RKEEPERGUID';

const unitId = `${testIdPrefix}-unit`;
const chainId = `${testIdPrefix}-chain`;
const groupId = `${testIdPrefix}-group`;

export const rkeeperUnit: RequiredId<CrudApi.CreateUnitInput> = {
  ...unitFixture.createRkeeperUnit,
  id: unitId,
  groupId,
  chainId,
  externalId: 'EXTERNAL-RESTAURANT-ID',
};

export const rkeeperUnitProduct: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}-unitproduct`,
  externalId: rkeeperProductGuid,
};

export const rkeeperUnitProduct2: RequiredId<CrudApi.CreateUnitProductInput> = {
  ...productFixture.unitProductInputBase,
  id: `${testIdPrefix}-unitproduct2`,
  externalId: rkeeperProductGuid + '2',
};

export const processedDish: Dish = {
  price: 50000,
  active: true,
  id: 9991000114,
  guid: '999e3ab3-86a0-48d9-a9a9-f4e0c9fbce68',
  name: 'Pr√≥ba ital',
};

export const businessEntity: RKeeperBusinessEntityInfo = {
  chainId,
  groupId,
  unitId,
};

export const createChain: RequiredId<CrudApi.CreateChainInput> = {
  ...chainFixture.chainBase,
  id: chainId,
};

export const createGroup: RequiredId<CrudApi.CreateGroupInput> = {
  ...groupFixture.groupBase,
  id: groupId,
  chainId,
};

export const rkeeperOrder = {
  objectid: yellowRestaurantId,
  remoteId: yellowRestaurantId,
  order_type: 1,
  pay_type: 0,
  pay_online_type: 0,
  delivery_time: '2122-02-03T12:37:32',
  client: {
    phone: null,
    email: null,
    ln: 'Testln',
    fn: 'Testfn',
  },
  order: [
    {
      id: '1000114',
      type: 'd',
      qnt: 1000,
    },
  ],
};

export const yellowUnit = {
  ...rkeeperUnit,
  externalId: yellowRestaurantId,
  pos: {
    ...rkeeperUnit.pos,
    rkeeper: {
      ...(rkeeperUnit?.pos?.rkeeper || {}),
      rkeeperUsername: yellowRkeeperUsername,
      rkeeperPassword: yellowRkeeperPassword,
      endpointUri: rkeeperEndpoint,
    } as CrudApi.RKeeper,
  },
} as CrudApi.Unit;

export const orderInput: CrudApi.CreateOrderInput = {
  userId: 'USER_ID',
  unitId: yellowUnit.id,
  items: [
    {
      quantity: 5,
      productId: 'PRODUCT ID',
      statusLog: [
        {
          userId: 'USER_ID',
          status: CrudApi.OrderStatus.none,
          ts: 1627909024677,
        },
      ],
      sumPriceShown: {
        taxSum: 316.98,
        currency: 'HUF',
        tax: 27,
        priceSum: 1491,
        pricePerUnit: 298.2,
      },
      productName: {
        hu: 'PRODUCT NAME HU',
      },
      priceShown: {
        taxSum: 318.9,
        currency: 'HUF',
        tax: 27,
        priceSum: 1500,
        pricePerUnit: 300,
      },
      variantId: 'VARIANT ID',
      variantName: {
        en: 'VARIANT NAME EN',
      },
    },
  ],
  sumPriceShown: {
    taxSum: 633.96,
    currency: 'HUF',
    tax: 27,
    priceSum: 2982,
    pricePerUnit: 298.2,
  },
  place: {
    table: '01',
    seat: '01',
  },
  orderMode: CrudApi.OrderMode.instant,
  servingMode: CrudApi.ServingMode.inplace,
  archived: false,
  paymentMode: {
    method: CrudApi.PaymentMethod.cash,
    type: CrudApi.PaymentType.cash,
  },
};

export {
  yellowRestaurantId,
  yellowRkeeperPassword,
  yellowRkeeperUsername,
  rkeeperEndpoint,
};
