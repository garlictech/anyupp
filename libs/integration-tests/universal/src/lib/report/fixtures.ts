import { DateTime } from 'luxon';

import { timezoneBudapest } from '@bgap/shared/utils';

const dateAgo = (days: number) =>
  DateTime.fromISO(new Date().toISOString(), {
    zone: timezoneBudapest,
  })
    .minus({ days })
    .toJSDate()
    .toString();

export const reportUsersFixture = [
  {
    id: 'anonym_1',
    email: 'testuser+anonymuser1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'anonym_2',
    email: 'testuser+anonymuser2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'anonym_3',
    email: 'testuser+anonymuser3@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_1',
    email: 'testuser+a1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_2',
    email: 'testuser+a2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'email_3',
    email: 'testuser+a3@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'facebook_1',
    email: 'testuser+fb1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'facebook_2',
    email: 'testuser+fb2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'google_1',
    email: 'testuser+g1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'google_2',
    email: 'testuser+g2@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'signinwithapple_1',
    email: 'testuser+apple1@anyupp.com',
    ucd: new Date().getTime(),
  },
  {
    id: 'signinwithapple_2',
    email: 'testuser+apple2@anyupp.com',
    ucd: new Date().getTime(),
  },
];

export const fourWeeksOrdersFixture = [
  {
    createdAt: dateAgo(1),
    userId: 'anonym_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(2),
    userId: 'anonym_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(3),
    userId: 'anonym_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(4),
    userId: 'anonym_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(5),
    userId: 'email_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(6),
    userId: 'email_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(7),
    userId: 'email_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(8),
    userId: 'email_3',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(9),
    userId: 'facebook_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(10),
    userId: 'facebook_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
];

export const thisYearOrdersFixture = [
  ...fourWeeksOrdersFixture,
  {
    createdAt: dateAgo(11),
    userId: 'facebook_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(42),
    userId: 'google_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(43),
    userId: 'google_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(44),
    userId: 'signinwithapple_1',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
  {
    createdAt: dateAgo(45),
    userId: 'signinwithapple_2',
    sumPriceShown: { currency: 'HUF', priceSum: 1000 },
  },
];
