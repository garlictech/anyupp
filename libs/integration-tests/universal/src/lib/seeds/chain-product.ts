import { CrudSdk } from '@bgap/crud-gql/api';
import { CreateChainProductInput } from '@bgap/domain';
import { resultTap } from './seed.util';

export const createTestChainProduct = (
  input: CreateChainProductInput,
  crudSdk: CrudSdk,
) =>
  crudSdk
    .CreateChainProduct({ input })
    .pipe(resultTap('CHAIN PRODUCT create', input.id!));

export const deleteTestChainProduct = (id: string, crudSdk: CrudSdk) =>
  crudSdk
    .DeleteChainProduct({ input: { id } })
    .pipe(resultTap('CHAIN PRODUCT delete', id));
