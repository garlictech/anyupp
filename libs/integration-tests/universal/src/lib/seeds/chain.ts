import * as CrudApi from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestChain = (
  input: CrudApi.CreateChainInput,
  crudSdk: CrudApi.CrudSdk,
) => crudSdk.CreateChain({ input }).pipe(resultTap('CHAIN create', input.id!));

export const deleteTestChain = (id: string, crudSdk: CrudApi.CrudSdk) =>
  crudSdk.DeleteChain({ input: { id } }).pipe(resultTap('CHAIN delete', id));
