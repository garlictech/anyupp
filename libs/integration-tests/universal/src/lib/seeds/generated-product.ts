import { CrudApi, CrudApiMutationDocuments } from '@bgap/crud-gql/api';
import {
  crudBackendGraphQLClient,
  executeMutation,
} from '@bgap/shared/graphql/api-client';
import { of } from 'rxjs';
import { catchError } from 'rxjs/operators';
import { resultTap } from './seed.util';

export const createTestGeneratedProduct = (
  input: CrudApi.CreateGeneratedProductInput,
) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.CreateGeneratedProductMutation
  >(CrudApiMutationDocuments.createGeneratedProduct, {
    input,
  }).pipe(resultTap('GENERATED_PRODUCT create', input.id!));

export const deleteTestGeneratedProduct = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<
    CrudApi.DeleteGeneratedProductMutation
  >(CrudApiMutationDocuments.deleteGeneratedProduct, {
    input: { id },
  }).pipe(
    resultTap('GENERATED_PRODUCT delete', id),
    catchError(() => of('NO_PROBLEM')),
  );
