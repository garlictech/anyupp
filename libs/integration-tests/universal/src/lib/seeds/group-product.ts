import { CrudSdk } from '@bgap/crud-gql/api';
import { CreateGroupProductInput } from '@bgap/domain';
import { resultTap } from './seed.util';

export const createTestGroupProduct = (
  input: CreateGroupProductInput,
  crudSdk: CrudSdk,
) =>
  crudSdk
    .CreateGroupProduct({ input })
    .pipe(resultTap('GROUP PRODUCT create', input.id!));

export const deleteTestGroupProduct = (id: string, crudSdk: CrudSdk) =>
  crudSdk
    .DeleteGroupProduct({ input: { id } })
    .pipe(resultTap('GROUP PRODUCT delete', id));
