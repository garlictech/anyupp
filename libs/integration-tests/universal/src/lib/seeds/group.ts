import * as CrudApi from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestGroup = (
  input: CrudApi.CreateGroupInput,
  crudSdk: CrudApi.CrudSdk,
) => crudSdk.CreateGroup({ input }).pipe(resultTap('GROUP create', input.id!));

export const deleteTestGroup = (id: string, crudSdk: CrudApi.CrudSdk) =>
  crudSdk.DeleteGroup({ input: { id } }).pipe(resultTap('GROUP delete', id));
