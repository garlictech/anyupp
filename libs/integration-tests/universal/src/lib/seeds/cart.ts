/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { cartSeed } from '../fixtures/cart';
import { tap, catchError } from 'rxjs/operators';
import { throwError } from 'rxjs';
import {
  crudBackendGraphQLClient,
  executeMutation,
} from '@bgap/shared/graphql/api-client';
import { CrudApi, CrudApiMutationDocuments } from '@bgap/crud-gql/api';
import { resultTap } from './seed.util';

export const createTestCart = (input: Partial<CrudApi.CreateCartInput>) =>
  executeMutation(crudBackendGraphQLClient)<CrudApi.CreateCartMutation>(
    CrudApiMutationDocuments.createCart,
    { input },
  ).pipe(resultTap('CART create', input.id!));

export const deleteTestCart = (id: string) =>
  executeMutation(crudBackendGraphQLClient)<CrudApi.DeleteCartMutation>(
    CrudApiMutationDocuments.deleteCart,
    {
      input: { id },
    },
  ).pipe(resultTap('CART delete', id));
