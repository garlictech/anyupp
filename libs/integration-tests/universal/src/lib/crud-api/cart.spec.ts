import * as CrudApi from '@bgap/crud-gql/api';
import { cartSeed } from '../fixtures/cart';
import {
  AuthenticatdGraphQLClientWithUserId,
  createAuthenticatedCrudGraphQLClient,
  executeQuery,
} from '@bgap/shared/graphql/api-client';
import { combineLatest } from 'rxjs';
import { switchMap } from 'rxjs/operators';
import { createTestCart, deleteTestCart } from '../seeds/cart';
import { testAdminUsername, testAdminUserPassword } from '../fixtures/user';

describe('getCart test', () => {
  let authHelper: AuthenticatdGraphQLClientWithUserId;

  beforeAll(async () => {
    authHelper = await createAuthenticatedCrudGraphQLClient(
      testAdminUsername,
      testAdminUserPassword,
    ).toPromise();
    console.warn(authHelper.userAttributes);

    await combineLatest([
      // CleanUP
      deleteTestCart(),
    ])
      .pipe(
        switchMap(() =>
          // Seeding
          combineLatest([createTestCart()]),
        ),
      )
      .toPromise();
  }, 15000);

  it('successful query execution', done => {
    executeQuery(authHelper.graphQlClient)<CrudApi.GetCartQuery>(
      CrudApi.getCart,
      { id: cartSeed.cart_01.id },
    ).subscribe({
      next(x) {
        expect(x.getCart?.id).toEqual(cartSeed.cart_01.id);
        done();
      },
    });
  }, 15000);
  it('should return null for a not existing item', done => {
    executeQuery(authHelper.graphQlClient)<CrudApi.GetCartQuery>(
      CrudApi.getCart,
      { id: cartSeed.cartId_NotExisting },
    ).subscribe({
      next(x) {
        expect(x.getCart).toBeNull();
        done();
      },
    });
  }, 10000);
  it('should throw error without id as input', done => {
    executeQuery(authHelper.graphQlClient)<CrudApi.GetCartQuery>(
      CrudApi.getCart,
    ).subscribe({
      error(e) {
        expect(e).toMatchSnapshot();
        done();
      },
    });
  }, 15000);
});
