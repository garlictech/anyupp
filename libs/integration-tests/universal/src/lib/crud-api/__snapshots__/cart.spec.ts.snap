// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCart test should throw error without id as input 1`] = `
Object {
  "code": "UserLambdaValidationException",
  "message": "PreTokenGeneration failed with error Network error: Response not successful: Received status code 403.",
  "name": "UserLambdaValidationException",
}
`;
